src/main/scala/com/target/data_validator/validator/ColumnBased.scala

```diff
@@ -10,11 +10,20 @@ import org.apache.spark.sql.catalyst.expressions.Expression
 import org.apache.spark.sql.catalyst.expressions.aggregate.Max
 import org.apache.spark.sql.types._
 
+import scala.collection.mutable.ListBuffer
+import scala.math.abs
+
 abstract class ColumnBased(column: String, condTest: Expression) extends CheapCheck {
   override def select(schema: StructType, dict: VarSubstitution): Expression = condTest
 
   // ColumnBased checks don't have per row error details.
   def hasQuickErrorDetails: Boolean = false
+
+  // calculates and returns the pct error as a string
+  def calculatePctError(expected: Double, actual: Double, formatStr: String = "%4.2f%%"): String = {
+    val pct = abs(((expected - actual) * 100.0) / expected)
```

Comment:
There are a few places where you're doing something to guard against div/0. I think you should move that guard here.

Reply:
The guards that I was using was to check for validation status (pass/fail), and not for div/0. Hence I have not removed them.
 But thanks for catching the div/0 issue. Now I have handled the issue in the method `calculatePctError`.